tools/clang/test/CodeGenSPIRV/type.enum.hlsl

```diff
@@ -0,0 +1,102 @@
+// Run: %dxc -T ps_6_0 -E main
+
+//CHECK:      %First = OpVariable %_ptr_Private_int Private %int_0
+//CHECK-NEXT: %Second = OpVariable %_ptr_Private_int Private %int_1
+//CHECK-NEXT: %Third = OpVariable %_ptr_Private_int Private %int_3
+//CHECK-NEXT: %Fourth = OpVariable %_ptr_Private_int Private %int_n1
+enum Number {
+  First,
+  Second,
+  Third = 3,
+  Fourth = -1,
+};
+
+//CHECK:      %a = OpVariable %_ptr_Private_int Private
+//CHECK-NEXT: %b = OpVariable %_ptr_Workgroup_int Workgroup
+//CHECK-NEXT: %c = OpVariable %_ptr_Uniform_type_AppendStructuredBuffer_ Uniform
+
+//CHECK:      [[second:%\d+]] = OpLoad %int %Second
+//CHECK-NEXT:                   OpStore %a [[second]]
+static ::Number a = Second;
+groupshared Number b;
+AppendStructuredBuffer<Number> c;
+
+void testParam(Number param) {}
+void testParamTypeCast(int param) {}
+
+void main() {
+//CHECK:      [[a:%\d+]] = OpLoad %int %a
+//CHECK-NEXT:              OpStore %foo [[a]]
+  int foo = a;
+
+//CHECK:      [[fourth:%\d+]] = OpLoad %int %Fourth
+//CHECK-NEXT:                   OpStore %b [[fourth]]
+  b = Fourth;
+
+//CHECK:          [[c:%\d+]] = OpAccessChain %_ptr_Uniform_int %c %uint_0
+//CHECK-NEXT: [[third:%\d+]] = OpLoad %int %Third
+//CHECK-NEXT:                  OpStore [[c]] [[third]]
+  c.Append(Third);
+
+  Number d;
+//CHECK:      [[d:%\d+]] = OpLoad %int %d
+//CHECK-NEXT:              OpSelectionMerge %switch_merge None
+//CHECK-NEXT:              OpSwitch [[d]] %switch_default 0 %switch_0 1 %switch_1
+  switch (d) {
+    case First:
+      d = Second;
+      break;
+    case Second:
+      d = First;
+      break;
+    default:
+      d = Third;
+      break;
+  }
+
+//CHECK:      [[fourth:%\d+]] = OpLoad %int %Fourth
+//CHECK-NEXT:                   OpStore %e [[fourth]]
+  static ::Number e = Fourth;
```

Comment:
Why are you using `static` everywhere?

Reply:
I tested
```
static ::Number a = Second;
groupshared Number b;
AppendStructuredBuffer<Number> c;
```

and in `main()`,
```
  Number d;
  static ::Number e = Fourth;
```

I just want to check many variable scope.
Is there a specific scope (or keyword) you want to test?